lib/libvfio-user.c

```diff
@@ -1943,4 +1966,20 @@ vfu_sg_is_mappable(vfu_ctx_t *vfu_ctx, dma_sg_t *sg)
     return dma_sg_is_mappable(vfu_ctx->dma, sg);
 }
 
+EXPORT void
+vfu_migr_done(vfu_ctx_t *vfu_ctx, int err)
+{
+    assert(vfu_ctx != NULL);
+    assert(vfu_ctx->migr_trans_pending);
+
+    if (vfu_ctx->migr_trans_msg != NULL) {
```

Comment:
when can this be null ?

Reply:
When the no_reply bit is set in the source message.

Reply:
Hmm I think I'd rather always stash the msg, so we can refer to it, then respect the no_reply here as needed.